### DISCUSSION TITLE
When a virus attacks a bottles prefix but the windows programs are stored outside that prefix,will they still be infected?

### DISCUSSION THREAD

**Discussion by ghost** [bottlesdevs/Bottles#354] [Q&A]
I store my windows applications somewhere else than directly in the bottles prefixes cause I like that more.
So my first question is,if I launch a windows program and that is a virus and is outside of my prefix,will it then attack the prefix in which I launched it?
The other question is,when a prefix is attacked by a virus does that virus spread over the other prefix or my whole system or just is in that one prefix?
And will only the programs that are stored in the infected bottle be infected or also programs that are stored outside but get launched in that bottle?
And if a bottle is infected,is there any way to restore it without the virus?

**Comment by mirkobrombin** [comment]
This seems more like a question of how Wine works. Anyway I try to give you an answer but for more details I ask you to write on the [WineHQ forum.](https://forum.winehq.org)

> So my first question is, if I launch a windows program and that is a virus and is outside of my prefix, will it then attack the prefix in which I launched it?

It depends on the type of virus, if it is an infectious virus it can spread and infect all files like wildfire, regardless of whether they are in the prefix or in the system. It does not matter even if the virus starts from the system or from the prefix because where you access, the virus accesses. However, keep in mind that a virus acts only when a wineserver is running, moreover, as for software compatibility, viruses must also be compatible in order to work, as Wine is not a real Windows operating system.

> The other question is, when a prefix is ​​attacked by a virus does that virus spread over the other prefix or my whole system or just is in that one prefix?

Answered above.

> And will only the programs that are stored in the infected bottle be infected or also programs that are stored outside but get launched in that bottle?

Same.

> And if a bottle is infected, is there any way to restore it without the virus?

You can enable experimental bottle [versioning](https://docs.usebottles.com/bottles/versioning) and then create a restore point to restore when you need it. But as I told you, every virus is different and this could also infect restore points.
